server/workflowarchive/archived_workflow_server.go

```diff
@@ -299,6 +300,20 @@ func (w *archivedWorkflowServer) RetryArchivedWorkflow(ctx context.Context, req
 			}
 		}
 
+		for _, podName := range podsToReset {
```

Comment:
I was gonna say that this may make sense to place into a helper function, but I see that https://github.com/argoproj/argo-workflows/pull/7988#discussion_r815240441 _explicitly_ moved the `kubeClient` logic out of the helper function (when retry was introduced for archived workflows in #7988).

So this matches existing behavior with `podsToDelete` above. If we refactored these into the Controller (per above comment), this would go away though 🤔